node-launchpad/src/components/popup/connection_mode.rs

```diff
@@ -305,7 +211,248 @@ impl Component for ChangeConnectionModePopUp {
             buttons_layer[1],
         );
 
-        // We render now so the borders are on top of the other widgets
+        pop_up_border
+    }
+
+    fn draw_confirm_change(
+        &mut self,
+        f: &mut crate::tui::Frame<'_>,
+        layer_zero: Rect,
+        layer_one: Rc<[Rect]>,
+    ) -> Paragraph {
+        // layer zero
+        let pop_up_border = Paragraph::new("").block(
+            Block::default()
+                .borders(Borders::ALL)
+                .title(" Confirm & Reset ")
+                .bold()
+                .title_style(Style::new().fg(VIVID_SKY_BLUE))
+                .padding(Padding::uniform(2))
+                .border_style(Style::new().fg(VIVID_SKY_BLUE)),
+        );
+        clear_area(f, layer_zero);
+
+        // split into 3 parts, paragraph, dash, buttons
+        let layer_two = Layout::new(
+            Direction::Vertical,
+            [
+                // for the text
+                Constraint::Length(9),
```

Comment:
Is it possible move this code into a common function for both `port_range.rs` and `connection_mode.rs`. It think there are a lot of moving components here and keeping them in sync might be hard. For e.g., port_range contains `Constraint::Length(8)` instead of `Constraint::Length(9)`.

Reply:
Indeed. I've tried with a helper module, but in the other hand we wanted self contained scenes/screens. I understand that this confirmation dialog is the same in the flows for the different types of connections, but they reside in different flow steps.

You would suggest a helper module with this function then?

Reply:
Fair enough, we can skip this one!